lib/GILGen/GILGenExpr.hpp

```diff
@@ -123,6 +123,126 @@ struct GILGenExpr : public ASTVisitor<GILGenExpr, gil::Value> {
         // Use bitcast for these situations
         return ctx.buildBitcast(destGilType, sourceValue)->getResult(0);
     }
+
+    gil::Value visitBinaryOpExpr(BinaryOpExpr *expr)
+    {
+        using namespace glu::ast;
+
+        // Get the token kind for the operator
+        TokenKind opKind = expr->getOperator().getKind();
+
+        // Special case for short-circuit logical operators (&& and ||)
+        if (opKind == TokenKind::andOpTok || opKind == TokenKind::orOpTok) {
+            // Generate code for the left operand first
+            gil::Value leftValue = visit(expr->getLeftOperand());
+
+            // Create a basic block that will contain the result
+            gil::BasicBlock *resultBB = ctx.buildBB("logical.result");
+
+            // Create the basic block for evaluating the right operand
+            gil::BasicBlock *evalRightBB = ctx.buildBB(
+                opKind == TokenKind::andOpTok ? "and.right" : "or.right"
+            );
+
+            // For AND: only evaluate right side if left side is true
+            // For OR: only evaluate right side if left side is false
+            if (opKind == TokenKind::andOpTok) {
+                // AND: If left is false, jump to result with "false", otherwise
+                // evaluate right operand
+                llvm::SmallVector<gil::Value, 1> falseArgs
+                    = { leftValue }; // Pass left value as "false"
+                llvm::SmallVector<gil::Value, 0>
+                    noArgs; // No args needed for right evaluation
+
+                ctx.buildCondBr(
+                    leftValue, evalRightBB, resultBB, noArgs, falseArgs
+                );
+
+                // Evaluate right operand
+                ctx.positionAtEnd(evalRightBB);
+                gil::Value rightValue = visit(expr->getRightOperand());
+
+                // Pass the result of the right evaluation to the result block
+                llvm::SmallVector<gil::Value, 1> rightResultArgs
+                    = { rightValue };
+                ctx.buildBr(resultBB);
+
+                // Position at the result block to continue
+                ctx.positionAtEnd(resultBB);
+
+                // Return an appropriate value based on the argument passed to
+                // the result block This would normally be handled by a phi node
+                // in the result block For now, return rightValue since we don't
+                // have proper block arguments yet
+                return rightValue;
```

Comment:
Return the argument, I think we have it? You can create a Value with BasicBlock and argument index 0